ui/packages/catalog-admin-module-sources/src/main/webapp/js/model/Source.js

```diff
@@ -57,29 +67,29 @@ define([
                     this.get("disabledConfigurations").remove(configuration);
                 } else if (configuration === this.get("currentConfiguration")) {
                     this.stopListening(configuration);
+                    this.stopPolling();
                     this.set("currentConfiguration", undefined);
                 }
             },
             setCurrentConfiguration: function (configuration) {
                 var sync = 'sync';
                 var statusUpdate = 'status:update-';
-                this.set({
-                    currentConfiguration: configuration
-                });
                 this.set({currentConfiguration: configuration});
 
                 var pid = configuration.id;
-                var statusModel = new Status.Model(pid);
-                statusModel.on(sync, function () {
-                    wreqr.vent.trigger(statusUpdate + pid, statusModel);
-                });
 
-                var options = {
-                    delay: 30000
-                };
+                var statusModel = this.get('statusModel');
```

Comment:
Is this updating it to avoid making multiple status models?

Reply:
Exactly. It would create a new model every time `setCurrentConfiguration` was called. And the old model would still be happily polling away 😄 I changed the source only uses a single model, and in turn, a single poller that can be cleaned up when the source is removed